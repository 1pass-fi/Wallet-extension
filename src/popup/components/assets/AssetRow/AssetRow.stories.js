import AssetRow from '.'

AssetRow.displayName = 'AssetRow'

export default {
  title: 'component/AssetRow',
  component: AssetRow,
  args: {
    isGrey: false,
    isKoiWallet: true,
    name: 'Test Asset Row', 
    isRegistered: true,
    earnedKoi: 1,
  }
}

const Template = (args) => <AssetRow {...args} />

export const Default = Template.bind({})
