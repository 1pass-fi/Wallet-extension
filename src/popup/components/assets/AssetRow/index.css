.asset-row-container {
  width: 100%;
  display: flex;
  align-items: center;
  padding: 0 30px 0 10px;
  height: 92px;
  box-sizing: border-box;
  color: #171753;
}

.asset-row-function-icons {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  cursor: pointer;
}

.asset-row-logo-icon {
  width: 38px;
  height: 38px;
  min-width: 38px;
  min-height: 38px;
  margin-left: 12px;
}

.asset-name {
  margin-left: 20px;
  font-weight: 600;
  font-size: 18px;
  letter-spacing: 0.03em;
  line-height: 20px;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  flex-grow: 2;
}

.koi-earned {
  min-width: 91px;
  width: 91px;
  margin: 34px 38px 34px 10px;
  font-size: 15px;
  font-weight: 400;
  line-height: 20px;
  letter-spacing: 0.03em;
}

.asset-row-function-icon {
  padding: 5px 0 5px 0;
}

.register-button {
  min-width: 97px;
  width: 97px;
  margin: 34px 38px 34px 10px;
  display: flex;
  flex-direction: row;
  align-items: center;
  padding-left: 8px;
  height: 16px;
  border: 1px solid #171753;
  box-sizing: border-box;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.16);
  border-radius: 2px;
  height: 24px;
  font-family: 'Catamaran', sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  letter-spacing: 0.03em;
}
.register-button  .register-button-icon {
  width: 18px;
  height: 18px;
  margin-right: 10px;
}

.assets .fish-icon {
  width: 14.4px;
  height: 10.44px;
}
