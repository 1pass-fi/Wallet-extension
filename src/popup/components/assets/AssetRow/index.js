import React from 'react'
import PropTypes from 'prop-types'

import ShareIcon from 'img/share-icon.svg'
import GalleryIcon from 'img/gallery-icon.svg'
import Fish from 'img/fish.svg'
import AlternativeWalletIcon from 'img/alternative-wallet-icon.svg'
import RearrangePadsIcon from 'img/rearrange-pads-icon.svg'
import './index.css'

const propTypes = {
  isGrey: PropTypes.bool,
  isKoiWallet: PropTypes.bool,
  name: PropTypes.string,
  isRegistered: PropTypes.bool,
  earnedKoi: PropTypes.number,
}

const WalletIcon = ({ isKoiWallet }) => (
  <div className="asset-row-logo-icon">
    { isKoiWallet ?
      <Fish />
      :
      <AlternativeWalletIcon />}
  </div>
)

const EarnedKoi = ({ isRegistered, earnedKoi }) => (
  isRegistered ?
    <div className="koi-earned">
      {earnedKoi} KOI earned
    </div>
    :
    <button className="register-button">
      <div className="register-button-icon">
        <div className='assets fish-icon'><Fish /></div>
      </div>
      <span>Register</span>
    </button>
)

const Actions = ({ isRegistered, viewblockUrl, galleryUrl }) => {

  const handleCreateTab = (to) => {
    chrome.tabs.create({
      url: to
    })
  }

  return (
    <div className="asset-row-function-icons">
      { isRegistered && <div onClick={() => handleCreateTab(viewblockUrl)}><ShareIcon className="asset-row-function-icon" /></div>}
      <div onClick={() => handleCreateTab(galleryUrl)}><GalleryIcon className="asset-row-function-icon" /></div>
    </div>
  )
}

const AssetRow = ({ isGrey, isKoiWallet, name, isRegistered, earnedKoi, viewblockUrl, galleryUrl }) => {
  return (
    <div className="asset-row-container" style={{ background: isGrey ? '#EEEEEE' : '#fff' }}>
      <div className="asset-row-rearrange-icon">
        <RearrangePadsIcon />
      </div>
      <WalletIcon isKoiWallet={isKoiWallet} />
      <div className="asset-name">{name}</div>
      <EarnedKoi isRegistered={isRegistered} earnedKoi={earnedKoi} />
      <Actions isRegistered={isRegistered} viewblockUrl={viewblockUrl} galleryUrl={galleryUrl} />
    </div>
  )
}

AssetRow.propTypes = propTypes

export default AssetRow
