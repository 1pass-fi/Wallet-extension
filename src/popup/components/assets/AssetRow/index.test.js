import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'

import AssetRow from '.'

describe('Test for  AssetRow component', () => {
  describe('Unregistered', () => {
    describe('Render without crashing', () => {
      it('renders correctly', () => {
        const { container } = render(<AssetRow isGrey={true} isKoiWallet={true} name="Asset Name" isRegistered={false} earnedKoi={0} />)
        expect(container).toMatchSnapshot('unregistered')
      })
    })
  })

  describe('Registered', () => {
    describe('Render without crashing', () => {
      it('renders correctly', () => {
        const { container } = render(<AssetRow isGrey={false} isKoiWallet={true} name="Asset Name" isRegistered={true} earnedKoi={0} />)
        expect(container).toMatchSnapshot('registered')
      })
    })
  })
})
