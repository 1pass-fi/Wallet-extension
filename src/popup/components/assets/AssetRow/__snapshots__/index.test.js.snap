// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for  AssetRow component Registered Render without crashing renders correctly: registered 1`] = `
<div>
  <div
    class="asset-row-container"
    style="background: rgb(255, 255, 255);"
  >
    <div
      class="asset-row-rearrange-icon"
    >
      <svg />
    </div>
    <div
      class="asset-row-logo-icon"
    >
      <svg />
    </div>
    <div
      class="asset-name"
    >
      Asset Name
    </div>
    <div
      class="koi-earned"
    >
      0
       KOI earned
    </div>
    <div
      class="asset-row-function-icons"
    >
      <svg
        class="asset-row-function-icon"
      />
      <svg
        class="asset-row-function-icon"
      />
    </div>
  </div>
</div>
`;

exports[`Test for  AssetRow component Unregistered Render without crashing renders correctly: unregistered 1`] = `
<div>
  <div
    class="asset-row-container"
    style="background: rgb(238, 238, 238);"
  >
    <div
      class="asset-row-rearrange-icon"
    >
      <svg />
    </div>
    <div
      class="asset-row-logo-icon"
    >
      <svg />
    </div>
    <div
      class="asset-name"
    >
      Asset Name
    </div>
    <button
      class="register-button"
    >
      <div
        class="register-button-icon"
      >
        <svg />
      </div>
      <span>
        Register
      </span>
    </button>
    <div
      class="asset-row-function-icons"
    >
      <svg
        class="asset-row-function-icon"
      />
    </div>
  </div>
</div>
`;
